spectre_analyses/tasks.py

```diff
@@ -0,0 +1,159 @@
+"""divik task definition for the purpose of use through Celery
+
+Copyright 2018 Spectre Team
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+from contextlib import contextmanager
+from functools import partial
+import json
+import os
+import pickle
+import re
+import shutil
+import sys
+
+import celery
+from celery.utils.log import get_task_logger
+
+import spdata.reader
+import spdata.types as ty
+
+from spectre_analyses.celery import app
+import matlab_hooks as mh
+
+
+FILESYSTEM_ROOT = os.path.abspath(os.sep)
+DATA_ROOT = os.path.join(FILESYSTEM_ROOT, 'data')
+STATUS_PATHS = {
+    'all': FILESYSTEM_ROOT,
+    'done': DATA_ROOT,
+    'processing': os.path.join(FILESYSTEM_ROOT, 'temp'),
+    'failed': os.path.join(FILESYSTEM_ROOT, 'failed')
+}
+Name = str
+Path = str
+DISALLOWED_CHARACTERS = "[^a-zA-Z0-9_-]"
+
+
+def _data_path(dataset_name: Name) -> Path:
+    dataset_name = re.sub(DISALLOWED_CHARACTERS, '_', dataset_name)
+    return os.path.join(DATA_ROOT, dataset_name, 'text_data', 'data.txt')
+
+
+def _load_data(dataset_name) -> ty.Dataset:
+    path = _data_path(dataset_name)
+    with open(path) as infile:
+        return spdata.reader.load_txt(infile)
+
+
+@contextmanager
+def _open_analysis(dataset_name: str, algorithm_name: str, analysis_name: str):
+    analysis_root = os.path.join(
+        STATUS_PATHS['processing'],
+        dataset_name,
+        algorithm_name,
+        analysis_name,
+    )
+    os.makedirs(analysis_root)
```

Comment:
Isn't it necessary to check if the directories don't already exist, or it is guaranteed they're unique each time? Or does the makedirs already take care of that?

Reply:
`os.makedirs` fails if directory exists :)